synapse/exc.py

```diff
@@ -361,3 +362,9 @@ class FatalErr(SynErr):
     pass
 
 class LmdbLock(SynErr): pass
+
+def reprexc(e):
+    text = repr(e)
+    if isinstance(e, SynErr):
+        text = e.get('mesg', text)
+    return text
```

Comment:
Did you see @vEpiphyte's comment about this implementation? Looks like you might have missed it by accident? https://github.com/vertexproject/synapse/pull/4336#discussion_r2137868493